tools/database-converter/punishments-converter.ts

```diff
@@ -0,0 +1,241 @@
+// Needed for FS
+global.Config = {nofswriting: false};
+
+import SQL from 'sql-template-strings';
+import * as sqlite from 'sqlite';
+import {FS} from '../../lib/fs';
+
+type SQLStatement = import('sql-template-strings').SQLStatement;
+type PunishmentEntry = import('../../server/punishments').PunishmentEntry;
+type PunishmentsDatabaseResponse = import('../../server/punishments').PunishmentsDatabaseResponse;
+type RoomPunishmentsDatabaseResponse = import('../../server/punishments').RoomPunishmentsDatabaseResponse;
+type SharedIpsDatabaseResponse = import('../../server/punishments').SharedIpsDatabaseResponse;
+type IpBanlistDatabaseResponse = import('../../server/punishments').IpBanlistDatabaseResponse;
+
+const PUNISHMENT_FILE = 'config/punishments.tsv';
+const ROOM_PUNISHMENT_FILE = 'config/room-punishments.tsv';
+const SHAREDIPS_FILE = 'config/sharedips.tsv';
+const IPBANLIST_FILE = 'config/ipbans.txt';
+
+const USERID_REGEX = /^[a-z0-9]+$/;
+
+export class PunishmentsConverter {
+	from: string;
+	to: string;
+	constructor(from: string, to: string) {
+		this.from = from;
+		this.to = to;
+	}
+	async convert() {
+		if (this.to === 'sqlite') {
+			const database = await PunishmentsConverterSqlite.databasePromise;
+			database.exec(`DELETE FROM punishments; DELETE FROM room_punishments; DELETE FROM shared_ips; DELETE FROM ip_banlist`).then(async () => {
+				// tslint:disable-next-line: no-floating-promises
+				PunishmentsConverterSqlite.convert();
+			}).catch(err => {
+				throw err;
+			});
+		} else if (this.to === 'tsv') {
+			PunishmentsConverterTsv.convert().catch(err => {
```

Comment:
yeah... why are you not just returning `PunishmentsConverterTsv.convert()` and letting the caller handle the error as desired?